lib/rubocop/cop/solidus/spree_t_decprecated.rb

```diff
@@ -0,0 +1,43 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Solidus
+      # This cop finds Spree.t method calls and replaces them with the I18n,t method call
+      # This cop is needed as the Spree.t version has been deprecated in future version.
+      #
+      #
+      # @example EnforcedStyle: bar (default)
+      #   # bad
+      #   Spree.t(:bar)
+      #
+      #   # good
+      #   I18n.t(:bar)
+      #
+      #   # good
+      #   good_bar_method(args)
+      #
+      class SpreeTDecprecated < Base
+        extend AutoCorrector
+        MSG = 'Use I18n.t instead of Spree.t which has been deprecated in future versions.'
+
+        RESTRICT_ON_SEND = %i[t].freeze
+
+        # @!method spree_t?(node)
+        def_node_matcher :spree_t?, <<~PATTERN
+          (send ($...) :t ...)
+        PATTERN
+
+        def on_send(node)
+          return unless spree_t?(node)
+
+          return unless spree_t?(node).include?(:Spree)
+
+          add_offense(node) do |corrector|
+            corrector.replace(node, node.source.gsub('Spree', 'I18n'))
```

Comment:
What is the purpose of this `replace`? 
Does it replace the code or just propose the correct code?

Reply:
If we have corrector then we can allow that cop to suggest and do auto corrections

Reply:
But this is wrong. Replacing the Spree with I18n is not enough. It also requires adding the `spree` scope as a parameter.

E.g.

Spree.t('user_addresses.success_update') -> I18n.t('user_addresses.success_update', scope: 'spree')

Reply:
Maybe we can simply remove it.

Reply:
I can update it